ethcore/src/snapshot/account.rs

```diff
@@ -62,53 +63,50 @@ impl CodeState {
 
 // walk the account's storage trie, returning an RLP item containing the
 // account properties and the storage.
-pub fn to_fat_rlp(acc: &BasicAccount, acct_db: &AccountDB, used_code: &mut HashSet<H256>) -> Result<Bytes, Error> {
+pub fn to_fat_rlps(acc: &BasicAccount, acct_db: &AccountDB, used_code: &mut HashSet<H256>, max_storage_items: usize) -> Result<Vec<Bytes>, Error> {
```

Comment:
Yes, exactly. This also ensures there won't be 12.01MB full chunks to begin with. If the chunk is 3MB and partial account record is another 3MB it gets written to the next chunk.